docs/guides/advanced-use-case/cross-namespace-repository.md

```diff
@@ -0,0 +1,681 @@
+      storageClassName: "standard"
+      resources:
+        requests:
+          storage: 1Gi
+```
+
+Let's create the StatefulSet we have shown above.
+
+```bash
+$ kubectl apply -f https://github.com/stashed/docs/raw/{{< param "info.version" >}}/docs/examples/guides/advanced-use-case/cross-namespace-repo/statefulset.yaml
+service/headless created
+statefulset.apps/stash-demo created
+```
+
+Now, wait for the pods of the StatefulSet to go into the `Running` state. You can verify the Status of the pods by ececuting the following command,
+
+```bash
+$ kubectl get pod -n dev
+NAME           READY   STATUS    RESTARTS   AGE
+stash-demo-0   1/1     Running   0          42s
+stash-demo-1   1/1     Running   0          40s
+stash-demo-2   1/1     Running   0          36s
+```
+
+Verify that the sample data has been generated in `/source/data` directory for `stash-demo-0` , `stash-demo-1` and `stash-demo-2` pod respectively using the following commands,
+
+```bash
+$ kubectl exec -n dev stash-demo-0 -- cat /source/data/data.txt
+stash-demo-0
+$ kubectl exec -n dev stash-demo-1 -- cat /source/data/data.txt
+stash-demo-1
+$ kubectl exec -n dev stash-demo-2 -- cat /source/data/data.txt
+stash-demo-2
+```
+
+### Prepare Backend
+
+We are going to store our backed up data into a GCS bucket. We have to create a Secret with necessary credentials and a Repository CRD to use this backend. We will create the Secret and Repository in `stash` namespace. With the cross-namespace Repository support, Stash has the ability to backup and restore data in a different namespace.
+
+If you want to use a different backend, please read the respective backend configuration doc from [here](/docs/guides/backends/overview.md).
+
+> For GCS backend, if the bucket does not exist, Stash needs `Storage Object Admin` role permissions to create the bucket. For more details, please check the following [guide](/docs/guides/backends/gcs.md).
+
+**Create Secret:**
+
+Let's create a secret called `gcs-secret` with access credentials to our desired GCS bucket,
+
+```bash
+$ echo -n 'changeit' > RESTIC_PASSWORD
+$ echo -n '<your-project-id>' > GOOGLE_PROJECT_ID
+$ cat /path/to/downloaded-sa-key.json > GOOGLE_SERVICE_ACCOUNT_JSON_KEY
+$ kubectl create secret generic -n stash gcs-secret \
+    --from-file=./RESTIC_PASSWORD \
+    --from-file=./GOOGLE_PROJECT_ID \
+    --from-file=./GOOGLE_SERVICE_ACCOUNT_JSON_KEY
+secret/gcs-secret created
+```
+
+Now, we are ready to backup our workload's data to our desired backend.
+
+**Create Repository:**
+
+Now, create a `Repository` using this secret. Below is the YAML of `Repository` crd we are going to create,
+
+```yaml
+apiVersion: stash.appscode.com/v1alpha1
+kind: Repository
+metadata:
+  name: gcs-repo
+  namespace: stash
+spec:
+  backend:
+    gcs:
+      bucket: stash-testing
+      prefix: /cross-namespace/data/sample-statefulset
+    storageSecretName: gcs-secret
+  usagePolicy:
+    allowedNamespaces:
+      from: All
+```
+
```

Comment:
Notice the `spec.usagePolicy` section. Here, we are allowing all namespaces to refer this repository. You can restrict this capability to a particular namespace. Please, check the following guide from [here]() for more details.